rados/rados_write_op_setxattr.go

```diff
@@ -0,0 +1,60 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <rados/librados.h>
+// #include <stdlib.h>
+//
+import "C"
+
+import (
+	"unsafe"
+)
+
+type writeOpSetXattrStep struct {
+	withoutUpdate
+	withoutFree
+	// The c pointer utilizes the Go byteslice data and no free is needed.
+
+	// Inputs
+
+	value []byte
+
+	// C arguments
+
+	cValue    *C.char
+	cValueLen C.size_t
+}
+
+func newWriteOpSetXattrStep(value []byte) *writeOpSetXattrStep {
+	return &writeOpSetXattrStep{
+		value:     value,
+		cValue:    (*C.char)(unsafe.Pointer(&value[0])), // TODO: must be pinned
+		cValueLen: C.size_t(len(value)),
+	}
+}
+
+// SetXattr sets an xattr.
+//  PREVIEW
+//
+// Implements:
+//  void rados_write_op_setxattr(rados_write_op_t write_op,
+//                               const char * name,
+//                               const char * value,
+//                               size_t value_len)
+func (w *WriteOp) SetXattr(name string, value []byte) {
+	s := newWriteOpSetXattrStep(value)
+	w.steps = append(w.steps, s)
+
+	cName := C.CString(name)
+	defer C.free(unsafe.Pointer(cName))
+
+	C.rados_write_op_setxattr(
+		w.op,
+		cName,
+		s.cValue,
+		s.cValueLen,
```

Comment:
```suggestion
		(*C.char)(unsafe.Pointer(&value[0])),
		C.size_t(len(value)),
```

and then you can remove all the other code... 🙂

Reply:
That's indeed a lot better, thanks!

Reply:
Hehe, sorry for reviewing all your code away. 🙈
But I always say: less code is better code. 😉

Reply:
Hah, sure thing!